src/rust/bitbox02/src/backup.rs

```diff
@@ -51,3 +53,50 @@ pub fn check() -> Result<CheckData, Error> {
         err => Err(err),
     }
 }
+
+#[derive(Copy, Clone, Default)]
+struct Backup(bitbox02_sys::Backup);
+
+#[derive(Copy, Clone, Default)]
+struct BackupData(bitbox02_sys::BackupData);
+
+impl zeroize::DefaultIsZeroes for Backup {}
+impl zeroize::DefaultIsZeroes for BackupData {}
+
+pub struct RestoreData {
+    pub seed: zeroize::Zeroizing<Vec<u8>>,
+    // unix timestamp, UTC.
+    pub birthdate: u32,
+    // unix timestamp, UTC.
+    pub timestamp: u32,
+    pub name: String,
+}
+
+pub fn restore_from_directory(dir: &str) -> Result<RestoreData, RestoreError> {
+    let mut backup = zeroize::Zeroizing::new(Backup {
+        ..Default::default()
+    });
+    let mut backup_data = zeroize::Zeroizing::new(BackupData {
```

Comment:
nice. I was going to say, isn't `backup_data.0.seed[..backup_data.0.seed_length as _].to_vec()` below leaves a copy behind somewhere in stack memory but this zeroizing erases it on drop. is that correct?

Reply:
I think so.

It is not obvious if `backup_data.0.seed[..backup_data.0.seed_length as _].to_vec()` internally makes copies or if it goes straight to the vector in heap though. I guess I could first make the Vector and then copy into it, making it very explicit.